src/path_manager.c

```diff
@@ -721,6 +749,135 @@ static void handle_mptcp_event(struct l_genl_msg *msg, void *user_data)
         };
 }
 
+/**
+ * @brief Verify that MPTCP is enabled at run-time in the kernel.
+ *
+ * Check that MPTCP is supported in the kernel by opening a socket
+ * with the @c IPPROTO_MPTCP protocol.
+ *
+ * @note @c IPPROTO_MPTCP is supported by the "MPTCP upstream"
+ *       kernel.
+ */
+static bool check_mptcp_socket_support(void)
+{
+#if !HAVE_DECL_IPPROTO_MPTCP
+# define IPPROTO_MPTCP 262
+#endif  // !HAVE_DECL_IPPROTO_MPTCP
+
+        int const fd = socket(AF_INET, SOCK_STREAM, IPPROTO_MPTCP);
+
+        // An errno other than EINVAL is unexpected.
+        if (fd != -1)
+                close(fd);
+        else if (errno != EINVAL)
+                l_error("Unable to confirm MPTCP socket support.");
+
+        return fd != -1;
+}
```

Comment:
Further simplify to avoid redundant file descriptor check:
```c
￼         // An errno other than EINVAL is unexpected.
￼         if (fd != -1) {
￼                 close(fd);
                 return true;
￼         } else if (errno != EINVAL) {
￼                 l_error("Unable to confirm MPTCP socket support.");
         }
￼
￼         return false;
￼ ```